toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/WorldScaleSceneView.kt

```diff
@@ -310,3 +278,210 @@ public fun WorldScaleSceneView(
         }
     }
 }
+
+/**
+ * Checks the permissions required for the [WorldScaleSceneView] to function and requests them if necessary.
+ * If [requestCameraPermissionAutomatically] or [requestLocationPermissionAutomatically] are set to false,
+ * the permissions will not be requested automatically and the initialization will fail if the permissions are not granted.
+ *
+ * @return A [MutableState] that will be true when the permissions have been requested and the result has been received.
+ * @since 200.7.0
+ */
+@Composable
+private fun requestPermissionsOrFail(
```

Comment:
Since this function returns a remembered state variable, I think we should prefix it with "remember", based on [convention](https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#naming-composable-functions-that-remember-the-objects-they-return). How about:
```suggestion
private fun rememberPermissionsGranted(
```